include/ignition/gui/qml/PlottingInterface.qml

```diff
@@ -0,0 +1,538 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+import QtQuick 2.9
+import QtCharts 2.2
+import QtQuick.Controls 2.2
+import QtQuick.Controls.Styles 1.4
+import QtQuick.Controls.Material 2.1
+import QtQuick.Layouts 1.3
+import Qt.labs.platform 1.0
+import "qrc:/qml"
+
+Rectangle
+{
+  id : main
+
+  Layout.minimumWidth: 600
+  Layout.minimumHeight: 600
+  anchors.fill: parent
+  color: (Material.theme == Material.Light) ? Material.color(Material.Grey,Material.Shade100) : "transparent"
+
+  property var charts: ({})
+  property int mainChartID: 1
+
+  property int idIncrementor: 0
+
+  property bool multiChartsMode: false
+
+  // Horizonal Layout to hold multi charts (small charts)
+  Rectangle {
+      id: rowCharts
+      width: parent.width
+      height: (multiChartsMode) ? 150 : 0
+      color: (Material.theme == Material.Light) ? Material.color(Material.Grey,Material.Shade200)
+                                                : Material.color(Material.BlueGrey, Material.Shade800)
+      // Make it Scrolable
+      ScrollView {
+          anchors.fill: parent
+          ScrollBar.horizontal.policy: ScrollBar.AsNeeded
+          ScrollBar.vertical.policy: ScrollBar.AlwaysOff
+          clip: true
+          // Horizontal Layout for the Charts
+          Row {
+              id:rowChartsLayout
+              anchors.fill: parent
+              spacing: 10
+          }
+      }
+  }
+
+  // Vertical Layout to hold Main Charts (1 or 2 charts)
+  Column {
+      id : chartsLayout
+
+      width: parent.width
+      anchors.topMargin: 10
+      anchors.top: rowCharts.bottom
+      anchors.bottom: parent.bottom
+      property int heightFactor: 0
+  }
+
+
+  Rectangle {
```

Comment:
Instead of using `Rectangle` + `Text` + `MouseArea`, consider using `ToolButton`, which has that all ready for you.

Reply:
ok I used the ToolButton in other buttons that has Rectangle + Text + MouseArea
but this one specifically i found that i will write a bigger code to style the ToolButton as i want than the Rectangle + Text + MouseArea :D, so i keep it